### PR TITLE
fix(react-badge): Presence badge should show Out of Office icon for Away+OOF and Offline+OOF statuses

### LINKED ISSUE #29101: [Bug]: Fluent v9 Persona no longer maps Out of Office + away/offline to the purple Out of Office icon
### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
System:
    OS: Linux 5.15 Debian GNU/Linux 12 (bookworm) 12 (bookworm)
    CPU: (2) x64 AMD EPYC 7763 64-Core Processor
    Memory: 6.01 GB / 7.75 GB
    Container: Yes
    Shell: 5.2.15 - /bin/bash
```

### Are you reporting Accessibility issue?

no

### Reproduction

https://codesandbox.io/s/persona-v8-v9-presence-mapping-7cclyg?file=/example.tsx

## Actual Behavior
Persona no longer maps the following statuses to the purple "Out of Office" icon
- `{ status: "away", outOfOffice: true }`
- `{ status: "offline", outOfOffice: true }`
- `{ status: "unknown", outOfOffice: true }`

## Expected Behavior
Persona should render these presence values as a purple "Out of Office" icon. 

In M365 and products there is no "Out of Office" status. This property is roughly determined by the presence of an automatic reply. There are two common behaviors when someone goes out of Office. 

1. User sets an automatic reply, and closes all communication apps. This causes their status to be `Offline` with `OutOfOffice == true` for the duration of their vacation. 
  - With v8 Persona their presence icon will be `Out of Office`
  - With v9 Persona their presence icon will be `Offline`
2. User sets an automatic reply and leaves communication apps open. This causes their status to be `Away` with `OutOfOffice == true` for the duration of their vacation. 
  - With v8 Persona their presence icon will be `Out of Office`
  - With v9 Persona their presence icon will be `Away - Out of Office`

## Explanation
This is a problem for a variety of reasons, for the Offline status there is no way to easily see the user is OOF. For the Away status, in practice no one really knows that outlined presence icons mean "Out of Office", they look for the purple icon. Furthermore, if we change this behavior, we will have mismatches in products that are on different versions of Fluent. 

Applications could solve this locally by doing the mapping, however, in this case this is something that is very important to be standard across the entire suite of products. We should not have many different mapping implementations. 

### Logs

_No response_

### Requested priority

High

### Products/sites affected

Monarch

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

### LINKED ISSUE #29121: [Persona, PresenceBadge]: Fluent v9 Presence now matches v8, how to get previous behavior?
## v9 off presence does not align with v8 (See below for call for collaboration)

This issue is a continuation of #29101. The bug in the issue mentioned before explains that v9 uses regular icons for the Away, Offline, and Unknown states. This is an issue since the new behavior might be confusing for users. There are a few solutions for this issue that each have their own drawbacks. Below is an image showing which icons are currently different.

image

### [Addressed by #29131]. Step 1: v9 has to match v8 behavior

v9 has to match how v8 behaves, this is so that apps in M365 currently using v9 do not have to create a mapping. This will help standardize the presence over all Microsoft apps including Teams. This is also an issue that is preferably solved on the library side rather than the application side. To achieve this, the following changes need to be made:

image

Notice that the OOF presence with a green rectangular border does not match. While this is arguably an issue, this is due to the available status being the default. When an oof flag is added but no presence, it results in available + oof flag. This is not an issue, but for sake of matching can be moved to the default status.

See [this hackMD](https://hackmd.io/@CfgNmipGTxmdTMlbMcDTbg/Hk79t0vAh) for more details.

### v9 Unkown + oof presence conflicts with v8 offline presence

On top of the Away/Offline/Unknown icons being different, we have an outstanding issue that requires more design guidance. As seen in the first image, the current unknown + oof presence matches the v8 offline presence. This is a conflict that will confuse users and needs to be addressed. One possibility is dropping the support of unknown since v8 didn't support it before, but this can affect apps currently using this presence.

# [Call for Collaboration] Step 2: current v9 behavior benefits outside users, how can we still offer that flexibility?

Making the changes from Step 1 means the previous behavior is no longer available. While on the internal partner side this is desirable, we want to give our outside users the option to use the previous icons. For this, we have two options that can address this. Option 1 will bloat a little more our API, but makes the change easier, while option 2 keeps the API as is, but requires a little more work for the user side.

### Option 1: Add a prop to enable the current v9 behavior and have the v8 behavior by default

Since we are modifying the mapping, we could add an option to turn this behavior off and keep the current one. To access this behavior, we could add an `enableExtraOOFIcons` (name open for suggestions) that would switch back to the current behavior. The resulting API would look like this:

```jsx
// Persona
<Persona
  presence={{
    enableExtraOOFIcons: true,
    status: 'away',
    outOfOffice: true
  }}
  name="Kevin Sturgis"
  secondaryText=" Away - Out of Office" 
/>

// PresenceBadge
<PresenceBadge enableExtraOOFIcons outOfOffice status="away" />
```

### Option 2: Update the behavior to match v8 and export the current icon mappings so the icon can be replaced through slots

We currently have internal mappings for the icons used in OOF to use the correct provided size. We could export these mappings and let the user override the default icon to achieve the previous behavior. The one drawback about doing so is that the user needs to provide the size for the icon. The usage would look like this:

```jsx
// Persona
<Persona
  presence={{
    status: 'away',
    outOfOffice: true,
    icon: presenceAway[size]
  }}
  name="Kevin Sturgis"
  secondaryText=" Away - Out of Office" 
/>

// PresenceBadge
<PresenceBadge icon={presenceAway[size]} enableExtraOOFIcons outOfOffice status="away" />
```

An example of the current mappings:

image

### PR DESCRIPTION
**Pull request by sopranopillow**
## Previous Behavior

image

## New Behavior

image

Presence icons are now exported in its package and in react-components.

## Related Issue(s)

#29101 mentions the issue this PR is fixing and #29121 explains the workaround implemented by this PR as well. If the previous behavior is required, please refer to the example in persona or the issue mentioned before for more details.
 
- Fixes #29101
- Fixes #29121

### CODE REVIEW

**Review by spmonahan — APPROVED** [review]

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29131/perf-test-react-components/Avatar-mount_base.html">639</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29131/perf-test-react-components/Avatar-mount.html">640</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29131/perf-test-react-components/Button-mount_base.html">321</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29131/perf-test-react-components/Button-mount.html">302</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29131/perf-test-react-components/Field-mount_base.html">1090</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29131/perf-test-react-components/Field-mount.html">1146</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29131/perf-test-react-components/FluentProvider-mount_base.html">680</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29131/perf-test-react-components/FluentProvider-mount.html">698</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29131/perf-test-react-components/FluentProviderWithTheme-mount_base.html">78</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29131/perf-test-react-components/FluentProviderWithTheme-mount.html">84</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29131/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">63</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29131/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">63</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29131/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">67</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29131/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">72</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>InfoButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29131/perf-test-react-components/InfoButton-mount_base.html">18</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29131/perf-test-react-components/InfoButton-mount.html">12</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29131/perf-test-react-components/MakeStyles-mount_base.html">864</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29131/perf-test-react-components/MakeStyles-mount.html">863</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29131/perf-test-react-components/Persona-mount_base.html">1733</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29131/perf-test-react-components/Persona-mount.html">1676</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29131/perf-test-react-components/SpinButton-mount_base.html">1362</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29131/perf-test-react-components/SpinButton-mount.html">1352</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports                                                                                            |     Baseline (minified/GZIP) |                           PR |                                                                                                                                                                                                     Change |
| :----------------------------------------------------------------------------------------------------------- | ---------------------------: | ---------------------------: | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------: |
| <samp>react-alert</samp> <br /> <abbr title='bundle-size/Alert.fixture.js'>Alert</abbr>                      |  `84.36 kB`<br />`23.094 kB` | `83.458 kB`<br />`22.856 kB` | `-902 B` <br />`-238 B`  |
| <samp>react-avatar</samp> <br /> <abbr title='bundle-size/Avatar.fixture.js'>Avatar</abbr>                   | `49.738 kB`<br />`15.631 kB` | `48.836 kB`<br />`15.406 kB` | `-902 B` <br />`-225 B`  |
| <samp>react-avatar</samp> <br /> <abbr title='bundle-size/AvatarGroupItem.fixture.js'>AvatarGroupItem</abbr> |  `64.39 kB`<br />`19.977 kB` | `63.488 kB`<br />`19.752 kB` | `-902 B` <br />`-225 B`  |
| <samp>react-badge</samp> <br /> <abbr title='bundle-size/PresenseBadge.fixture.js'>PresenceBadge</abbr>      |    `24.75 kB`<br />`8.96 kB` |  `23.852 kB`<br />`8.733 kB` | `-898 B` <br />`-227 B`  |
| <samp>react-persona</samp> <br /> <abbr title='bundle-size/Persona.fixture.js'>Persona</abbr>                | `56.629 kB`<br />`17.505 kB` | `55.727 kB`<br />`17.285 kB` | `-902 B` <br />`-220 B`  |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                             |          Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: |
| <samp>react-avatar</samp> <br /> <abbr title='bundle-size/AvatarGroup.fixture.js'>AvatarGroup</abbr>                                                                          |   `18.584 kB`<br />`7.448 kB` |
| <samp>react-badge</samp> <br /> <abbr title='bundle-size/Badge.fixture.js'>Badge</abbr>                                                                                       |   `25.793 kB`<br />`8.348 kB` |
| <samp>react-badge</samp> <br /> <abbr title='bundle-size/CounterBadge.fixture.js'>CounterBadge</abbr>                                                                         |   `26.694 kB`<br />`8.658 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr>              |  `69.328 kB`<br />`19.608 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `208.141 kB`<br />`59.349 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>                            |  `40.713 kB`<br />`13.509 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr>                                                         |    `6.541 kB`<br />`2.227 kB` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/DataGrid.fixture.js'>DataGrid</abbr>                                                                                 | `156.416 kB`<br />`43.567 kB` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/PrimitivesOnly.fixture.js'>Table (Primitives only)</abbr>                                                            |  `42.463 kB`<br />`13.255 kB` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/TableAsDataGrid.fixture.js'>Table as DataGrid</abbr>                                                                 | `129.276 kB`<br />`34.739 kB` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/TableSelectionOnly.fixture.js'>Table (Selection only)</abbr>                                                         |   `74.56 kB`<br />`20.053 kB` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/TableSortOnly.fixture.js'>Table (Sort only)</abbr>                                                                   |  `73.191 kB`<br />`19.649 kB` |
| <samp>react-tags-preview</samp> <br /> <abbr title='bundle-size/InteractionTag.fixture.js'>InteractionTag</abbr>                                                              |    `13.85 kB`<br />`5.626 kB` |
| <samp>react-tags-preview</samp> <br /> <abbr title='bundle-size/tag.fixture.js'>Tag</abbr>                                                                                    |   `29.604 kB`<br />`9.567 kB` |
| <samp>react-tags-preview</samp> <br /> <abbr title='bundle-size/TagGroup.fixture.js'>TagGroup</abbr>                                                                          |  `72.493 kB`<br />`21.628 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/5894f6b7c66621e2da61d792bfb9ba1215797cec'>5894f6b7c66621e2da61d792bfb9ba1215797cec</a></sub>
<div id="bundleSizeComment9423"/>

File: change/@fluentui-react-badge-3bd9df4e-4e19-4ff6-b2bf-accfb6602308.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (fabricteam) [inline change/@fluentui-react-badge-3bd9df4e-4e19-4ff6-b2bf-accfb6602308.json:1]: ## [🕵 fluentuiv9 Open the Visual Regressions report to inspect the affected screenshots]( https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-approval/index.html?organization=uifabric&project=fabricpublic&prId=29131&commitId=b9c2429d1eebfae74d96e57da07eb7cea73bfcc9&env=prod&clientType=FLUENTUI)


  <details><summary><b>PresenceBadge Converged - OOF status</b> 2 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|PresenceBadge Converged - OOF status.default - RTL.chromium.png|48| Changed|
|PresenceBadge Converged - OOF status.default.chromium.png|48| Changed|

</details>


  <details><summary><b>PresenceBadge Converged - inverted background</b> 3 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|PresenceBadge Converged - inverted background.default - High Contrast.default.chromium.png|269| Changed|
|PresenceBadge Converged - inverted background.default - Dark Mode.default.chromium.png|231| Changed|
|PresenceBadge Converged - inverted background.default.default.chromium.png|289| Changed|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

Comment (sopranopillow) [reply change/@fluentui-react-badge-3bd9df4e-4e19-4ff6-b2bf-accfb6602308.json:1]: changes are correct

**Review by miroslavstastny — APPROVED** [review]
It would be great if we could crop the `PresenceBadge Converged - OOF status` VR story to make it more visible.

**Comment by JustSlone** [comment]
We should add a way to achieve the old behavior (or really to force a particular icon to be displayed) in case the mapping doesn't work for someone, or they want to maintain the existing behavior.

File: change/@fluentui-react-badge-3bd9df4e-4e19-4ff6-b2bf-accfb6602308.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+  "type": "patch",
+  "comment": "fix: Match OOF icon mapping to v8.",
```
Comment (JustSlone) [inline change/@fluentui-react-badge-3bd9df4e-4e19-4ff6-b2bf-accfb6602308.json:3]:
```suggestion
  "comment": "fix: Presence badge should show Out of Office icon for away+OOF and Offline+OOF statuses.",
```
We should be more descriptive in the change

File: change/@fluentui-react-badge-3bd9df4e-4e19-4ff6-b2bf-accfb6602308.json [hunk 3]
```diff
@@ -0,0 +1,7 @@
...
+  "packageName": "@fluentui/react-badge",
+  "email": "esteban.230@hotmail.com",
```
Comment (JustSlone) [inline change/@fluentui-react-badge-3bd9df4e-4e19-4ff6-b2bf-accfb6602308.json:5]: shouldn't this be your microsoft e-mail?

Comment (sopranopillow) [reply change/@fluentui-react-badge-3bd9df4e-4e19-4ff6-b2bf-accfb6602308.json:5]: My account has both the microsoft one and my personal. Don't know how to change it, but if it's needed I can find it out.

**Pull request merged**